### ISSUE TITLE
Locked Tokens Support should have an option to request Dapper Labs as creator of Shared Account (2) 

### ISSUE THREAD

**Issue by laynelafrance** [onflow/flow-core-contracts#37]
### Issue To Be Solved

This would allow third parties to securely and reliably receive and manage locked tokens on a user's behalf. 

### Proposed Solution(s) 
Account Setup should have an option for the user agent to create only the User Account `account 1` and the ability to request the creation of the Shared Account `account 2` by Dapper Labs. This ensures Flow can safely and securely enforce token lock-ups without blocking community wallets from supporting people who hold locked Flow. 
 
Add the ability for a user agent to request the `flow token admin` (or some other Dapper Labs controlled account) creates `account 2` (so that Dapper Labs does not have to trust or check that `account 2` was created correctly by the user agent. 

Following initial account creation, the process and interactions should be identical. The user and custody/wallet provider's roles, abilities, and interactions should be the same.

* Details of the technical implementation @psiemens has given some thought to various options

@psiemens options:
> 1) Custodian is permitted to create the locked account, but only through a special method: FlowServiceAccount.createLockedAccount . This method would simply return the LockedTokens resource which the custodian can link into the unlocked account. We then know that the locked account was created properly
> 2) Custodian is not permitted to create the locked account and instead places a LockedTokenReceiver in the unlocked account. We then create the locked account at the time of token delivery and connect it to the unlocked account with the receiver.
> 3) The locked account is created automatically by the protocol when createAccount (AuthAccount(…)) is called. Every account receives a locked sidecar account by default

### Caveats and considerations for the future 
**Option 1 considerations:** 
Currently custodians have account creator capability
Anyone unregisteered locked account is un-trusted
We could modify service account to have account creator capability - would return token manager resource which could be linked into public account - input would be users pubkey
tl;dr service account is account creator via pub.fun calling restricted code

**Smart contract changes needed:**
* a pub.fun doesnt require giving custodians their own capability - they could just call pub.fun --> reduces contract/integration load 
* This **will not** remove earlier methods for creating locked account but will be an additional integration option 

**Other changes needed:**
* Write updated transactions
* Write contract changes
* Document this alternative account creation process
* Re-deploy service account code (adding pub.fun for creating account) 

**Option 2 considerations:**
Complicates matters to not have both accounts exist immediately 

**Option 3 considerations:** 
Not all accounts need to receive locked tokens and this would create hidden complexity

Reference Documentation: https://docs.onflow.org/token/staking/locked/setup/

###  Context

This would ensure the Blocto wallet could participate in supporting the initial token delivery.
